core/services/chainlink/config_general.go

```diff
@@ -345,6 +345,19 @@ func (g *generalConfig) StarkNetEnabled() bool {
 	return false
 }
 
+// TODO: Update this method once Aptos becomes public integration
+// BCI-3511: https://smartcontract-it.atlassian.net/browse/BCI-3511
+func (g *generalConfig) AptosEnabled() bool {
+	for _, c := range g.c.Aptos {
+		if c != nil {
+			if v, ok := (*c)["Enabled"]; ok {
+				return v == nil || *v.(*bool) // Enabled attribute for config is *bool
```

Comment:
do we need check if the typecast doesn't fail here?

Reply:
for loop plugins, the type would be bool pointer. 

```
type TOMLConfig struct {
	ChainID   *string
	FeederURL *config.URL
	// Do not access directly. Use [IsEnabled]
	Enabled *bool
	Chain
	Nodes Nodes
}

func (c *TOMLConfig) IsEnabled() bool {
	return c.Enabled == nil || *c.Enabled
}
```

good idea to do a type cast check

Reply:
maybe there is a more elegant way to handle it? since the interface for the function does not allow error to be thrown 🤔

Reply:
Why don't we expand the validation to check that this field can be cast to bool?

Reply:
@jmank88 could be good!
it's just one of the things i look for when I see casting to make sure something can't panic if the user does something weird

Reply:
thanks for the feedback @aalu1418 & @jmank88 added typecasting for *bool and bool

and using panic to make panic explicit instead of implicit